cinema/views.py

```diff
@@ -118,21 +132,23 @@
     max_page_size = 100
 
 
-class OrderViewSet(viewsets.ModelViewSet):
+class OrderViewSet(BaseViewSet):
     queryset = Order.objects.prefetch_related(
         "tickets__movie_session__movie", "tickets__movie_session__cinema_hall"
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
 
     def get_queryset(self):
         return Order.objects.filter(user=self.request.user)
 
-    def get_serializer_class(self):
-        if self.action == "list":
-            return OrderListSerializer
-
-        return OrderSerializer
-
-    def perform_create(self, serializer):
-        serializer.save(user=self.request.user)
+    # def get_serializer_class(self):
+    #     if self.action == "list":
+    #         return OrderListSerializer
+    #
+    #     return OrderSerializer
+    #
+    # def perform_create(self, serializer):
+    #     serializer.save(user=self.request.user)
```

Comment:
The `perform_create` method is commented out. If you need to automatically associate the order with the current user upon creation, consider uncommenting and using this method.